From 3c874c639808693c7e08338d7b95e84e8542c2d1 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 2 Mar 2023 14:47:56 +0200
Subject: [PATCH 48/54] Revert "s32cc: mmu: add adc0 entry"

This reverts commit 8df4677272d4f8333ed4e5c89b8a330b4ceafe63.

The ADC support in ATF is deleted as it's no longer needed for
identifying the RDB revision to apply a DDR fixup.

Issue: ALB-9722
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/include/s32_platform_def.h       | 3 ---
 plat/nxp/s32/s32_bl2_el3.c                    | 2 --
 plat/nxp/s32/s32g/include/s32g_platform_def.h | 2 +-
 plat/nxp/s32/s32r/include/s32r_platform_def.h | 4 ++--
 4 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/plat/nxp/s32/include/s32_platform_def.h b/plat/nxp/s32/include/s32_platform_def.h
index 26534fe1e..7d9fdb334 100644
--- a/plat/nxp/s32/include/s32_platform_def.h
+++ b/plat/nxp/s32/include/s32_platform_def.h
@@ -205,8 +205,5 @@
 #define S32_SCP_CH_MEM_SIZE	(128)
 #define S32_SCP_SCMI_MEM_SIZE	(S32_SCP_CH_MEM_SIZE * PLATFORM_CORE_COUNT)
 
-#define ADC0_BASE_ADDR		(0x401f8000U)
-#define ADC0_SIZE		(0x1000)
-
 #endif /* S32_PLATFORM_H */
 
diff --git a/plat/nxp/s32/s32_bl2_el3.c b/plat/nxp/s32/s32_bl2_el3.c
index 4040add1b..fc5cd17c4 100644
--- a/plat/nxp/s32/s32_bl2_el3.c
+++ b/plat/nxp/s32/s32_bl2_el3.c
@@ -226,8 +226,6 @@ static mmap_region_t s32_mmap[] = {
 			MT_DEVICE | MT_RW),
 	MAP_REGION_FLAT(I2C4_BASE_ADDR, MMU_ROUND_UP_TO_PAGE(I2C4_SIZE),
 			MT_DEVICE | MT_RW),
-	MAP_REGION_FLAT(ADC0_BASE_ADDR, MMU_ROUND_UP_TO_PAGE(ADC0_SIZE),
-			MT_DEVICE | MT_RW),
 	MAP_REGION_FLAT(OCOTP_BASE_ADDR, MMU_ROUND_UP_TO_PAGE(OCOTP_SIZE),
 			MT_DEVICE | MT_RW),
 	MAP_REGION_FLAT(SRAMC0_BASE_ADDR, MMU_ROUND_UP_TO_PAGE(SRAMC_SIZE),
diff --git a/plat/nxp/s32/s32g/include/s32g_platform_def.h b/plat/nxp/s32/s32g/include/s32g_platform_def.h
index 0cb1cdc55..915a16114 100644
--- a/plat/nxp/s32/s32g/include/s32g_platform_def.h
+++ b/plat/nxp/s32/s32g/include/s32g_platform_def.h
@@ -57,7 +57,7 @@
 #define BL31SSRAM_MAILBOX	(S32G_SSRAM_BASE)
 
 #define MAX_MMAP_REGIONS		40
-#define MAX_XLAT_TABLES			28
+#define MAX_XLAT_TABLES			27
 
 #if defined IMAGE_BL33
 #pragma warning "BL33 image is being built; you should configure it out."
diff --git a/plat/nxp/s32/s32r/include/s32r_platform_def.h b/plat/nxp/s32/s32r/include/s32r_platform_def.h
index 32fef5a67..b845e7b66 100644
--- a/plat/nxp/s32/s32r/include/s32r_platform_def.h
+++ b/plat/nxp/s32/s32r/include/s32r_platform_def.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021-2022 NXP
+ * Copyright 2021-2023 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -22,7 +22,7 @@
 #define S32_MPIDR_CPU_MASK_BITS	0x1
 
 #define MAX_MMAP_REGIONS		31
-#define MAX_XLAT_TABLES			26
+#define MAX_XLAT_TABLES			25
 
 #if defined IMAGE_BL33
 #pragma warning "BL33 image is being built; you should configure it out."
-- 
2.25.1

